tools/codegen/gen.py

```diff
@@ -243,13 +248,111 @@ def __call__(self, f: Union[StructuredNativeFunctions, NativeFunction]) -> List[
         else:
             assert_never(f)
 
+    def gen_structured_class_set_output(self, k: SchemaKind, parent_class: str, generate_super: bool) -> str:
+        if generate_super:
+            set_output_super = f"{parent_class}::set_output(output_idx, sizes, strides, options, names);"
+        else:
+            set_output_super = ""
+        return f"""
+void set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides,
+                TensorOptions options, DimnameList names) override {{
+    {self.gen_structured_class_set_output_body(k)}
+    if (!names.empty()) namedinference::propagate_names(outputs_[output_idx], names);
+    // super must happen after, so that downstream can use maybe_get_output
+    // to retrieve the output
+    {set_output_super}
+}}
+"""
+
+    def gen_structured_class_set_output_body(self, k: SchemaKind) -> str:
+        if k is SchemaKind.functional:
+            if self.dispatch_key == "Meta":
+                return """
+if (strides.empty()) {
+    outputs_[output_idx] = at::empty_meta(sizes, options);
+} else {
+    TORCH_INTERNAL_ASSERT(0, "not implemented yet");
+}
+"""
+            else:
+                expanded_topts = "optTypeMetaToScalarType(options.dtype_opt()), options.layout_opt(), " \
+                    "options.device_opt(), options.pinned_memory_opt()"
+                if self.dispatch_key == "CPU":
+                    empty_impl = "at::native::empty_cpu"
+                    empty_strided_impl = "at::native::empty_strided_cpu"
+                elif self.dispatch_key == "CUDA":
+                    empty_impl = "at::native::empty_cuda"
+                    empty_strided_impl = "at::native::empty_strided_cuda"
```

Comment:
[no action needed] calling back to a thread from previous codegen PRs, these names are more examples of symbols that I'd hoist out to global definitions and treat as part of the common data model that codegen and library are defined over.

Reply:
I briefly thought about doing some sort of platinum solution whereby we actually consult `native_functions.yaml` to find out what the post dispatch versions of empty are so that we can codegen them out correctly, even if someone modifies `native_functions.yaml`. But dumb and simple won the day for now.